packages/x-data-grid/src/hooks/features/filter/gridFilterSelector.ts

```diff
@@ -96,6 +106,41 @@ export const gridFilteredSortedRowIdsSelector = createSelectorMemoized(
   (filteredSortedRowEntries) => filteredSortedRowEntries.map((row) => row.id),
 );
 
+/**
+ * Get the ids to position in the current tree level lookup of the rows accessible after the filtering process.
+ * Does not contain the collapsed children.
+ * @category Filtering
+ * @ignore - do not document.
+ */
+export const gridExpandedSortedRowTreeLevelPositionLookupSelector = createSelectorMemoized(
```

Comment:
Did you already consider computing this lookup during the filtering as you did with the `filteredChildrenCountLookup` to remove one extra `O(n)` loop?

I am imagining the worst case with many unwrapped rows (for example using `defaultGroupingExpansionDepth="-1"`) might yield a slowdown, especially if the data size is more than usual.

What do you think?

Reply:
I did.
Thing is, I need both, a tree structure and the lookups that are flattened out to determine position in a specific tree level. That info cannot be added to the node itself
So, to get it to here a couple of other selectors would have to be updated (possibly with a breaking change)
The measurements that we made show that this selector does not create a big performance impact, so I am thinking of leaving it as is.
The processing time does not depend on the tree structure since it goes through a flat Id list